lib/index.js

```diff
@@ -75,8 +62,114 @@ function _getNSIndex(nsName, siteInfo) {
     return index;
 }
 
+/**
+ * Creates a namespace instance from namespace text or a namespace alias
+ *
+ * @param {string} text Namespace name text.
+ * @param {SiteInfo} siteInfo the site information.
+ * @returns {Namespace|undefined} a namespace or undefined if it wasn't found.
+ */
+Namespace.fromText = function(text, siteInfo) {
+    var index = _getNSIndex(text, siteInfo);
+    if (index !== undefined) {
+        return new Namespace(index, siteInfo);
+    }
+    return undefined;
+};
+
+/**
+ * Creates a namespace object for a `Main` namespace.
+ *
+ * @param {SiteInfo} siteInfo the site information.
+ * @returns {Namespace}
+ */
+Namespace.main = function(siteInfo) {
+    return new Namespace(0, siteInfo);
+};
+
+/**
+ * Checks whether namespace is `Special`.
+ *
+ * @returns {boolean}
+ */
+Namespace.prototype.isSpecial = function() {
+    return this._id === -1;
+};
+
+/**
+ * Checks whether namespace is `Main`.
+ *
+ * @returns {boolean}
+ */
+Namespace.prototype.isMain = function() {
+    return this._id === 0;
+};
+
+/**
+ * Checks whether namespace is `Talk`.
+ *
+ * @returns {boolean}
+ */
+Namespace.prototype.isTalk = function() {
+    return this._id === 1;
+};
+
+/**
+ * Checks whether namespace is `User`.
+ *
+ * @returns {boolean}
+ */
+Namespace.prototype.isUser = function() {
+    return this._id === 2;
+};
+
+/**
+ * Checks whether namespace is `User_Talk`.
+ *
+ * @returns {boolean}
+ */
+Namespace.prototype.isUserTalk = function() {
+    return this._id === 3;
+};
+
+/**
+ * Get the canonical name string for this namespace.
+ *
+ * @returns {string}
+ */
+Namespace.prototype.getNormalizedText = function() {
+    return this._siteInfo.namespaces[this._id + '']['*'].replace(/ /g, '_');
+};
+
+/**
+ * Returns the case parameter for the namespace
+ *
+ * @returns {string}
+ */
+Namespace.prototype.getCase = function() {
+    return this._siteInfo.namespaces[this._id + ''].case;
+};
+
+var regexCache = {};
+function _createInvalidTitleRegex(legalTitleChars) {
+    if (regexCache[legalTitleChars]) {
+        return regexCache[legalTitleChars];
+    }
+    // Any character not allowed is forbidden
+    regexCache[legalTitleChars] = new RegExp('[^' +
+            utils.convertByteClassToUnicodeClass(legalTitleChars) + ']' +
+            // URL percent encoding sequences interfere with the ability
+            // to round-trip titles -- you can't link to them consistently.
+            '|%[0-9A-Fa-f]{2}' +
+            // XML/HTML character references produce similar issues.
+            '|&[A-Za-z0-9\x80-\xff]+;' +
+            '|&#[0-9]+;' +
+            '|&#x[0-9A-Fa-f]+;');
+    return regexCache[legalTitleChars];
+}
+
 function _capitalizeTitle(result, siteInfo) {
-    if (siteInfo.namespaces[result.namespace].case === 'first-letter') {
+    if (result.namespace.getCase() === 'first-letter') {
```

Comment:
I thought the title capitalization was global, but the namespace capitalization could differ per namespace. [The docs](https://www.mediawiki.org/wiki/API:Siteinfo) say:

- In the `General` section: case: Whether the first letter in a title is case-insensitive ("first-letter") or case-sensitive ("case-sensitive"). See $wgCapitalLinks. 1.8+
- In the `Namespace` section: case: Either "first-letter" or "case-sensitive". If the first letter of the namespace name is capitalized (the default), then the value of this attribute will be "first-letter". Otherwise, the value will be "case-sensitive" in order to indicate that the namespace name in question intentionally does not use a capital first letter. Note: "case-insensitive" is reserved for future use 1.16+